File: TypeScript/TwoPointers/bestTime.ts
function maxProfit(prices: number[]): number {
    let left = 0;
    let profit = 0;

    for (let right = 1; right < prices.length; right++) {
        if (prices[right] < prices[left]) {
            left = right;
        } else {
            profit = Math.max(profit, prices[right] - prices[left]);
        }
    }

    return profit;
}

File: Python/TwoPointers/bestTime.py
def maxProfit(prices):
    left = 0  # ngày mua (giá thấp nhất hiện tại)
    profit = 0

    for right in range(1, len(prices)):  # ngày bán
        if prices[right] < prices[left]:
            left = right  # tìm được giá thấp hơn -> cập nhật ngày mua
        else:
            profit = max(profit, prices[right] - prices[left])

    return profit